server/src/main/java/com/aws/greengrass/cli/CLIService.java

```diff
@@ -165,8 +165,15 @@ private void registerIpcEventStreamHandlers() {
     @Override
     protected void install() {
         try {
-            Path clientArtifact = kernel.getNucleusPaths().unarchiveArtifactPath(new ComponentIdentifier(CLI_SERVICE,
-                    new Semver(Coerce.toString(getConfig().find(VERSION_CONFIG_KEY)))), CLI_CLIENT_ARTIFACT);
+            String version = Coerce.toString(getConfig().find(VERSION_CONFIG_KEY));
+            if (version == null) {
+                logger.atError().log(
+                        "Unable to find the CLI component version, aborting installation of CLI Client binary");
+                return;
```

Comment:
can you add a comment here on why we are returning instead of throwing an error

Reply:
I was thinking along these lines when I added "aborting installation of CLI Client binary" to this log message, but I can add in a comment on this whole section of the install method and why it returns, would probably be better I agree